### PR TITLE
feat(updating password):allow user to update password at any amount o…

### PR DESCRIPTION
**Pull request by Ruthuwamahoro**
#### What does this PR do?

Change password at x amount of time 

#### Description of Task to be completed?

- Sending a message to the email to update the password before the expiration date
- If the Notification is neglected and the date exceeds the expired date,he/she is prompted to reset the password

#### How should this be manually tested?

- set up the env according to your localhost database credentials
- use the api in the index.ts in routers to connect to the database
- use npm run dev to connect to the database

#### Any background context you want to provide?

-dummy data are provided before having the real data

#### What are the relevant pivotal tracker/Trello stories?

- id of the project
  #187300173

### CODE REVIEW

File: src/controllers/loginController.ts [hunk 1]
```diff
@@ -17,6 +19,24 @@ const loginController = async (req: Request, res: Response) => {
         'User not found or Invalid Credentials'
       );
     }
+    if (passwordExpired(user.lastTimePasswordUpdated)) {
+      return Database.User.findOne({
+        where: {
+          lastTimePasswordUpdated: user.lastTimePasswordUpdated,
+        },
+      }).then(() => {
+        if (user.roleId !== 'd290f1ee-6c54-4b01-90e6-d701748f0853') {
```
Comment (fab-ryan) [inline src/controllers/loginController.ts]: don't use id use name please

File: src/controllers/passwordExpirationcron.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { Op } from 'sequelize';
+import Database from '../database';
+import { passwordExpirationNotification } from '../utils/passwordExpirationCheck';
+
+const checkPasswordExpiration = async () => {
+  const getAllUsers = await Database.User.findAll({
+    where: {
+      roleId: {
+        [Op.ne]: 'd290f1ee-6c54-4b01-90e6-d701748f0853',
+      },
```
Comment (fab-ryan) [inline src/controllers/passwordExpirationcron.ts]: again use name

File: src/middleware/authentication.ts [hunk 3]
```diff
@@ -31,6 +39,29 @@ export const isAuthenticated = (
       if (isLogout) {
         return sendResponse(res, 401, null, 'You are not authorized');
       }
+      if (passwordExpired(user.lastTimePasswordUpdated)) {
+        return Database.User.findOne({
+          where: {
+            lastTimePasswordUpdated: user.lastTimePasswordUpdated,
+          },
+        }).then(() => {
+          if (user.roleId !== 'd290f1ee-6c54-4b01-90e6-d701748f0853') {
```
Comment (fab-ryan) [inline src/middleware/authentication.ts]: don't use id

**Review by fab-ryan — CHANGES_REQUESTED** [review]
change that don't use id

**Review by musabehonore — APPROVED** [review]

File: src/middleware/authentication.ts (hunk 3 above)
Comment (Ruthuwamahoro) [reply src/middleware/authentication.ts]: done

File: src/controllers/passwordExpirationcron.ts (hunk 2 above)
Comment (Ruthuwamahoro) [reply src/controllers/passwordExpirationcron.ts]: done

File: src/controllers/loginController.ts (hunk 1 above)
Comment (Ruthuwamahoro) [reply src/controllers/loginController.ts]: solved

**Review by Nadinefiona — APPROVED** [review]

**Review by fab-ryan — APPROVED** [review]

**Pull request merged**